jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/internal/AnyUserLoginService.java

```diff
@@ -0,0 +1,109 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security.siwe.internal;
+
+import java.util.function.Function;
+import javax.security.auth.Subject;
+
+import org.eclipse.jetty.security.DefaultIdentityService;
+import org.eclipse.jetty.security.IdentityService;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.security.UserIdentity;
+import org.eclipse.jetty.security.UserPrincipal;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Session;
+
+public class AnyUserLoginService implements LoginService
+{
+    private final String _realm;
+    private final LoginService _loginService;
+    private IdentityService _identityService;
+    private boolean _authenticateNewUsers;
+
+    public AnyUserLoginService(String realm)
+    {
+        this(realm, null);
+    }
+
+    public AnyUserLoginService(String realm, LoginService loginService)
+    {
+        _realm = realm;
+        _loginService = loginService;
+        _identityService = (loginService == null) ? new DefaultIdentityService() : null;
+    }
+
+    /**
+     * This setting is only meaningful if a wrapped {@link LoginService} has been set.
+     * <p>
+     * If set to true, any users not found by the wrapped {@link LoginService} will still
+     * be authenticated but with no roles, if set to false users will not be
+     * authenticated unless they are discovered by the wrapped {@link LoginService}.
+     * </p>
+     * @param authenticateNewUsers whether to authenticate users not found by a wrapping LoginService
```

Comment:
double negatives with exceptions is doing my head in. 
```suggestion
 * Configures the behavior for authenticating users not found by a wrapped {@link LoginService}.
 * <p>
 * This setting is only meaningful if a wrapped {@link LoginService} has been set.
 * </p>
 * <p>
 * If set to {@code true}, users not found by a wrapped {@link LoginService} will authenticated with no roles.
 * If set to {@code false}, only users found by a wrapped {@link LoginService} will be authenticated.
 * </p>
 * 
 * @param authenticateNewUsers whether to authenticate users not found by the wrapped {@link LoginService}
```

Reply:
Even that does my head in.   Isn't this just the `create` parameter of `getUserIdentity` ?  Call it so!

Reply:
I think `getUserIdentity` is a new Jetty 12 method. And I think this is a really strange method and possibly wrong, why does it assume the credentials is `null`.